src/mvc/Controller.php

```diff
@@ -184,4 +187,21 @@ public function end()
     {
         // empty
     }
+
+    /**
+     * Caches the rendered view and echoes the cached content.
+     *
+     * @param string $view          The view to render.
+     * @param array  $cacheOptions  An array of cache options:
+     *                  - 'key': The cache key to store the view content.
+     *                  - 'expiry': The expiration time for the cached content.
+     * @return void
+     */
+    public function cacheAndRender($view, $cacheOptions)
+    {
```

Comment:
You can do like this - 

```
ob_start();
$this->render($view);
$response = ob_get_clean();

$this->cache->store($cacheOptions['key'], $response, $cacheOptions['expiry']);
echo $response;
```

Please remove the changes you made in `render()` function